plasmapy/diagnostics/charged_particle_radiography.py

```diff
@@ -1115,118 +1263,161 @@ def max_deflection(self):
+        running `~plasmapy.diagnostics.charged_particle_radiography.Tracker`.
 
-        intensity : ndarray, shape (hbins, vbins)
-            The number of particles counted in each bin of the histogram.
-        """
+    size : `~astropy.units.Quantity`, shape ``(2, 2)``, optional
+        The size of the detector array, specified as the minimum
+        and maximum values included in both the horizontal and vertical
+        directions in the detector plane coordinates. Shape is
+        ``((hmin, hmax), (vmin, vmax))``. If not specified, the size will be
+        set to include all particles on the detector. Units must be convertable
+        to meters.
 
-        # Note that, at the end of the simulation, all particles were moved
-        # into the image plane.
+    bins : array of integers, shape ``(2)``
+        The number of bins in each direction in the format
+        ``(hbins, vbins)``.  The default is ``(200, 200)``.
 
-        # If ignore_grid is True, use the predicted positions in the absence of
-        # simulated fields
-        if ignore_grid:
-            x = self.x0
-        else:
-            x = self.x
+    ignore_grid: `bool`
+        If `True`, returns the intensity in the image plane in the absence
+        of simulated fields.
 
-        # Determine locations of points in the detector plane using unit
-        # vectors
-        xloc = np.dot(x - self.detector, self.det_hdir)
-        yloc = np.dot(x - self.detector, self.det_vdir)
-
-        if size is None:
-            # If a detector size is not given, choose lengths based on the
-            # dimensions of the grid
-            w = self.mag * np.max(
-                [
-                    np.max(np.abs(self.grid.pts0.to(u.m).value)),
-                    np.max(np.abs(self.grid.pts1.to(u.m).value)),
-                    np.max(np.abs(self.grid.pts2.to(u.m).value)),
-                ]
-            )
+    optical_density: `bool`
+        If `True`, return the optical density rather than the intensity
+
+        .. math::
+            OD = -log_{10}(Intensity/I_0)
 
-            # The factor of 5 here is somewhat arbitrary: we just want a
-            # region a few times bigger than the image of the grid on the
-            # detector, since particles could be deflected out
-            size = 5 * np.array([[-w, w], [-w, w]]) * u.m
+        where :math:`Intensity` is the simulation intensity on the
+        detector plane and :math:`I_0` is the intensity on the detector
+        plane in the absence of simulated fields. Default is `False`.
 
-        # Generate the histogram
-        intensity, h, v = np.histogram2d(
-            xloc, yloc, range=size.to(u.m).value, bins=bins
+        If the `Intensity` histogram contains zeros, `OD` will contain
+        -`~numpy.inf` values. These can be easily replaced with zeros
+        if desired for plotting using `~numpy.nan_to_num`
+
+        ``OD = np.nan_to_num(OD, neginf=0)``
```

Comment:
```suggestion
        where :math:`Intensity` is the simulation intensity on the
        detector plane and :math:`I_0` is the intensity on the detector
        plane in the absence of simulated fields. Default is `False`.
        If the :math:`Intensity` histogram contains zeros, then the
        corresponding values in :math:`OD` will be `numpy.inf`.  When
        plotting :math:`OD` the `~numpy.inf` values can be replaced
        using ``numpy.nan_to_num(OD, neginf=0, posinf=0)``.
```

The rendering of parameter definitions is very limited.  You basically need to keep everything in one paragraph for it to render correctly.

I also replaced `-numpy.inf` with `numpy.inf` since the OD expression is `-numpy.log10(I/I0)` which means `0` maps to `numpy.inf`.